oclint-scripts/build

```diff
@@ -29,8 +29,6 @@ def build_command(llvm_root, is_release, module_extras, source_path):
     cmd_build = cmake.builder(source_path)
     if is_release:
         cmd_build.release_build()
-    if environment.is_unix():
-        cmd_build.use_clang_compiler(llvm_root)
```

Comment:
I understand using system compiler might be the common use case on linux, but I need to consider another big group of users, which use OCLint on Mac. So using a clang version that is checked out directly from official repo and build the tool against it will help for Mac users.

Another thing to consider is that, OCLint rules heavily depend on the AST that is generated by clang. At times in the past, the system clang may not catch up with the changes that is required by OCLint. On Mac particular, sometimes, the clang compiler embedded inside Xcode could be different than the one published from LLVM community, which introduced even deeper confusions. So when I ship the release version, I need to build it with the local clang.

So, please help keep the default as using local clang compiler. We can tweak other scripts and make them more friendly to users who use system compilers.